x/tokenomics/token_logic_module/validator_distribution_test.go

```diff
@@ -211,59 +165,330 @@ func TestDistributeValidatorRewards_ErrorCases(t *testing.T) {
+		delegations              map[string][]stakingtypes.Delegation
+		rewardAmount             math.Int
+		expectedTransferCount    int
+		expectedValidatorRewards int64
+		expectedDelegatorRewards int64
+		opReason                 tokenomicstypes.SettlementOpReason
+		validation               func(*testing.T, *tokenomicstypes.ClaimSettlementResult)
+	}{
 		{
-			OperatorAddress: sample.ValOperatorAddressBech32(),
-			Tokens:          math.NewInt(333), // ~33.3% of total
-			Status:          stakingtypes.Bonded,
+			name: "success: mixed delegation amounts with equal self-bonded stakes",
+			validators: []stakingtypes.Validator{
+				createValidator(sample.ValOperatorAddressBech32(), 1_000_000),
+				createValidator(sample.ValOperatorAddressBech32(), 600_000),
+				createValidator(sample.ValOperatorAddressBech32(), 400_000),
+			},
+			rewardAmount:             math.NewInt(100_000),
+			expectedTransferCount:    5,      // 3 validators + 2 delegators
+			expectedValidatorRewards: 60_000, // 100k × 60% = 60,000 (self-bonded validator shares)
+			expectedDelegatorRewards: 40_000, // 100k × 40% = 40,000 (delegator shares)
+			opReason:                 tokenomicstypes.SettlementOpReason_TLM_GLOBAL_MINT_VALIDATOR_REWARD_DISTRIBUTION,
 		},
 		{
-			OperatorAddress: sample.ValOperatorAddressBech32(),
-			Tokens:          math.NewInt(333), // ~33.3% of total
-			Status:          stakingtypes.Bonded,
+			name: "success: equal self-bonded stakes receive equal rewards despite different total delegations",
+			validators: []stakingtypes.Validator{
+				createValidator(sample.ValOperatorAddressBech32(), 900_000),
+				createValidator(sample.ValOperatorAddressBech32(), 600_000),
+				createValidator(sample.ValOperatorAddressBech32(), 300_000),
+			},
+			rewardAmount:          math.NewInt(90_000),
+			expectedTransferCount: 5, // 3 validators + 2 delegators
+			opReason:              tokenomicstypes.SettlementOpReason_TLM_RELAY_BURN_EQUALS_MINT_VALIDATOR_REWARD_DISTRIBUTION,
+			validation: func(t *testing.T, result *tokenomicstypes.ClaimSettlementResult) {
+				// Verify equal validator rewards despite different total delegations
+				transfers := result.GetModToAcctTransfers()
+				validatorRewardMap := make(map[string]int64)
+				for _, transfer := range transfers {
+					if transfer.OpReason == tokenomicstypes.SettlementOpReason_TLM_RELAY_BURN_EQUALS_MINT_VALIDATOR_REWARD_DISTRIBUTION {
+						validatorRewardMap[transfer.RecipientAddress] = transfer.Coin.Amount.Int64()
+					}
+				}
+				// Each validator should get 15,000 (300k self-bonded out of 1.8M total)
+				for _, reward := range validatorRewardMap {
+					require.Equal(t, int64(15_000), reward)
+				}
+			},
 		},
 		{
-			OperatorAddress: sample.ValOperatorAddressBech32(),
-			Tokens:          math.NewInt(334), // ~33.4% of total
-			Status:          stakingtypes.Bonded,
+			name: "success: backward compatibility when no delegations found (falls back to validator-only)",
```

Comment:
I don't think this is a "backwards compatability" thing. It's simply a case where there are zero delegations. No?